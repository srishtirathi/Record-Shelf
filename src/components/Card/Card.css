.record-card {
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.3);
    width: 40px, padding 3rem;
    padding: 2rem;
    border-color: white;
    border-style: solid;
    border-width: 2rem;
}

.record-container {
    margin-top: 20px;
    padding: 2px 16px;
}

.song-img {
    margin-top: 20px;
    width: 200px;
    height: 150px;
}

@media screen and (max-width: 27rem) {
    .card {
        width: auto;
        padding-top: 20px;
    }
}