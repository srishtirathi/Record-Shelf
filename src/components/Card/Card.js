/* eslint-disable react/prop-types */
import React from 'react';
import './Card.css';

const Card = ({ name, artist, albumArtUrl }) => (
  <div className="record-card" data-testid="card">

    <img data-testid="record-img" src={albumArtUrl} className="song-img" alt="songImg" />
    <div className="record-container">
      <p>

        {name}
      </p>
      <p>

        {artist}
      </p>
    </div>

  </div>
);
export default Card;
