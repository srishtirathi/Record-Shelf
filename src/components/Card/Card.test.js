/* eslint-disable react/jsx-props-no-spreading */
import { render } from '@testing-library/react';
import React from 'react';
import {
  BrowserRouter,
} from 'react-router-dom';
import Card from './Card';

describe(Card.name, () => {
  const mockProps = {
    name: 'abc',
    artist: 'abc',
    albumArtUrl: 'abc',
  };
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('sanity check', () => {
    expect(1 + 1).toBe(2);
  });
  it('should match snapshot', () => {
    const { container } = render(<BrowserRouter><Card {...mockProps} /></BrowserRouter>);
    expect(container).toMatchSnapshot();
  });
});
