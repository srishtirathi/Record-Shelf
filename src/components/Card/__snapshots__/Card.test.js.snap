// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<div>
  <div
    class="record-card"
    data-testid="card"
  >
    <img
      alt="songImg"
      class="song-img"
      data-testid="record-img"
      src="abc"
    />
    <div
      class="record-container"
    >
      <p>
        abc
      </p>
      <p>
        abc
      </p>
    </div>
  </div>
</div>
`;
